src/bra_fs.cpp

```diff
@@ -347,40 +349,43 @@ bool file_set_add_dir(std::set<std::filesystem::path>& files) noexcept
     return true;
 }
 
-bool search(const std::filesystem::path& dir, const std::string& pattern, std::list<std::filesystem::path>& out_files) noexcept
+bool search(const std::filesystem::path& dir, const std::string& pattern, std::list<std::filesystem::path>& out_files, const bool recursive) noexcept
 {
     try
     {
         const std::regex r(pattern);
 
         // TODO: add a cli flag for fs::directory_options::skip_permission_denied
-        // TODO: add a cli flag for fs::directory_options::follow_directory_symlink
+        // TODO: add a cli flag for fs::directory_options::follow_directory_symlink (to replace symlink with the real file)
         for (const auto& entry : fs::directory_iterator(dir))
         {
-            fs::path ep = entry.path();
-            // TODO: dir to search only if it is recursive (-r)
+            fs::path   ep     = entry.path();
             const bool is_dir = dir_exists(ep);
             if (!(file_exists(ep) || is_dir))
                 continue;
 
             const std::string filename = ep.filename().string();
-            if (!std::regex_match(filename, r))
-                continue;
 
             if (is_dir)
             {
-                // bra_log_debug("Matched dir: %s", filename.c_str());
-
-                // TODO: if recursive...
-                //       actually would be better to use recursive_directory_iterator instead
+                // TODO: it might be better to use recursive_directory_iterator instead
+                //       of bra::fs::search being recursive itself.
+                if (!recursive)
+                {
+                    bra_log_info("Skip directory: %s", filename.c_str());
+                    continue;
+                }
```

Comment:
@coderabbitai create an issue relative to this highlighted code and TODO comment